File: pairwiseAndSum.py


n = int(input())

a = input().split()
a = [int(i) for i in a]
#print(a)
nonZero = [0 for i in range(30)]

for num in a:

    for i in range(30):
        andAnswer = num & (1<<i)
        if andAnswer !=0:
            nonZero[i]+=1

answer = 0
for i in range(30):
    answer += (nonZero[i]*(nonZero[i]-1)*(1<<i))

print(answer//2)

File: PairwiseAndSum.java
import java.util.Scanner;


public class PairwiseAndSum {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int a[] = new int[n]; int nonZero[] = new int[30];
        for(int i =0;i<n;i++){
            a[i] = sc.nextInt();

        }


        for(int i =0;i<n;i++){
            int num = a[i];
            for(int j =0;j<30;j++){
                int bit = num & (1<<j);
                if(bit != 0) nonZero[j]++;
            }

        }

        long answer=0;
        for(int j =0;j<30;j++){
            answer += ((long)nonZero[j]*((long)nonZero[j]-1L)*(1L<<j));
        }

        System.out.println(answer/2L);

    }
}
